src/dotnet/Microsoft.DotNet.Core.Sdk.targets

```diff
@@ -0,0 +1,100 @@
+    <_FullFrameworkReferenceAssemblyPaths>$(MSBuildThisFileDirectory)</_FullFrameworkReferenceAssemblyPaths>
+  </PropertyGroup>
+  
+  <UsingTask TaskName="MakeNetCoreRunnable" AssemblyFile="$(MicrosoftDotNetCoreBuildTasksDirectory)Microsoft.DotNet.Core.Build.Tasks.dll" />
+             
+  <PropertyGroup>
+    <CoreBuildDependsOn>
+      $(CoreBuildDependsOn);
+      GenerateDependencyFile;
+      GenerateRuntimeConfigurationFiles
+    </CoreBuildDependsOn>
+  </PropertyGroup>
+
+  <Target Name="GenerateDependencyFile"
+          Condition=" '$(GenerateDependencyFile)' == 'true'">
+    <!--
+    TODO: Is this the best way? Should we just tell CSC to output $(AssemblyName).dll directly?
+    -->
+    <Move SourceFiles="$(TargetPath)" DestinationFiles="$(TargetDir)/$(AssemblyName).dll" />
+
+    <!--
+    TODO: need to remove ProjectPath (to load up the project.json) and instead pass @(PackageDependency) 
+    -->
+    <MakeNetCoreRunnable ProjectPath="$(MSBuildProjectDirectory)"   
+                         Configuration="$(Configuration)"
+                         OutputPath="$(OutDir)"
+                         OutputType="$(OutputType)" />
+  </Target>
+
+  <Target Name="GenerateRuntimeConfigurationFiles"
+        Condition=" '$(GenerateRuntimeConfigurationFiles)' == 'true'">
+
+    <!-- TODO: Split up MakeNetCoreRunnable and implement this target-->
+    <!-- Putting this here to show the structure that we should use in production -->
+  </Target>
+
+  <!--
+    ============================================================
+                                        GetRunInformation
+
+    Returns the ProcessStart information that can be used to run this project.
+    ============================================================
+    -->
+
+  <Target Name="GetRunInformation"
+          DependsOnTargets="GenerateRunInformation"
+          Outputs="@(RunInformation)" />
+
+  <Target Name="GenerateRunInformation">
+    <ItemGroup>
+      <RunInformation Include="$(MSBuildProjectFullPath)">
+	    <!-- TODO: Need to get the Muxer path from MSBuild -->
+        <CommandName>dotnet</CommandName>
+        <Args>exec;$(TargetDir)/$(AssemblyName).dll</Args>
+      </RunInformation>
+    </ItemGroup>
+  </Target>
+
+  <!--
+    ============================================================
+                                        Publish
+
+    Override the "in-box" publish, which is very ClickOnce centric.
+    ============================================================
+    -->
+
+  <PropertyGroup>
+    <PublishDependsOn>
+      Build;
+      CopyPublishArtifacts
+    </PublishDependsOn>
+
+    <PublishDir Condition=" '$(PublishDir)' == '' ">$(OutDir)publish\</PublishDir>
+  </PropertyGroup>
+
+  <Target
+      Name="Publish"
+      DependsOnTargets="$(PublishDependsOn)" />
+
+  <Target Name="CopyPublishArtifacts">
+    <Message Text="TODO: Implement CopyPublishArtifacts" />
```

Comment:
this will flesh out later on

---
In reply to: [71799484](https://github.com/dotnet/cli/pull/3911#discussion_r71799484) [](ancestors = 71799484,71799150,71798669,71798420,71796387,71795755)